cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -191,7 +197,9 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) errors.AutoscalerError
 	// the newest node became available for the scheduler.
 	allNodesAvailableTime := GetAllNodesAvailableTime(readyNodes)
 	podsToReset, unschedulablePodsToHelp := SlicePodsByPodScheduledTime(allUnschedulablePods, allNodesAvailableTime)
-	ResetPodScheduledCondition(a.AutoscalingContext.ClientSet, podsToReset)
+	if *internalResetUnschedulablePodConfition {
```

Comment:
I think we should put the whole pod condition  related logic behind this flag (ie. including SlicePodsByPodScheduledTime), not just resetting. If we don't reset the condition the LastTransitionTime for unschedulable pod will never get updated and we will never do anything to help it.

On the other hand if we skip this whole part of logic at worst we will add an unnecessary node if the race condition with scheduler gets triggered, which I think is much smaller evil.